ksqldb-common/src/main/java/io/confluent/ksql/util/VertxUtils.java

```diff
@@ -40,7 +41,20 @@ public static void checkContext(final Context context) {
   }
 
   public static boolean isEventLoopAndSameContext(final Context context) {
-    return Context.isOnEventLoopThread() && context == Vertx.currentContext();
+    return Context.isOnEventLoopThread()
+        && (context == Vertx.currentContext()
+        || checkDuplicateContext(Vertx.currentContext(), context)
+        || checkDuplicateContext(context, Vertx.currentContext()));
+  }
+
+  private static boolean checkDuplicateContext(final Context context, final Context other) {
```

Comment:
*blinks*

I kinda want a method like this to check that the two inputs are duplicate regardless of the order.  Not sure if the refactoring is worth it; your call.